packages/cli/config/jest.js

```diff
@@ -20,6 +20,58 @@ const crypto = require('crypto');
 const glob = require('util').promisify(require('glob'));
 const { version } = require('../package.json');
 
+const notSupportedProjectAttrs = [
+  'bail',
+  'changedSince',
+  'changedFilesWithAncestor',
+  'collectCoverage',
+  'collectCoverageFrom',
+  'collectCoverageOnlyFrom',
+  'coverageDirectory',
+  'coverageProvider',
+  'coverageReporters',
+  'coverageThreshold',
+  'enabledTestsMap',
+  'expand',
+  'findRelatedTests',
+  'forceExit',
+  'json',
+  'lastCommit',
+  'logHeapUsage',
+  'listTests',
+  'maxConcurrency',
+  'maxWorkers',
+  'noStackTrace',
+  'nonFlagArgs',
+  'noSCM',
+  'notify',
+  'notifyMode',
+  'outputFile',
+  'onlyChanged',
+  'onlyFailures',
+  'passWithNoTests',
+  'projects',
+  'replname',
+  'reporters',
+  'runTestsByPath',
+  'silent',
+  'testFailureExitCode',
+  'testNamePattern',
+  'testPathPattern',
+  'testResultsProcessor',
+  'testSequencer',
+  'testTimeout',
+  'updateSnapshot',
+  'useStderr',
+  'verbose',
+  'watch',
+  'watchAll',
+  'watchman',
+  'watchPlugins',
+  'path',
+  'config',
+];
+
```

Comment:
Like that you keep the types in mind, but maybe this is a bit out of scope for solving the issue & this PR. You could add comments to reference the difference in types between `ProjectConfig` & `GlobalConfig` in case people this is refactored to TypeScript in the future. 

I see the risk that this approach breaks with future Jest config changes. In my opinion it make sense to get a warning if someone has a missconfiguration in Jest.

Reply:
Removed from this PR and added a commment 👍